From 7d8df0b54e464f996ae60776ca0185f03bc73a1d Mon Sep 17 00:00:00 2001
From: Yuvarani V <quic_yuvarani@quicinc.com>
Date: Wed, 15 Nov 2023 12:18:48 +0530
Subject: [PATCH] wifi: ath12k: Remove Transition Timeout value 11 map per D3.0

For compliance with IEEE802.11be D3.0, remove the mapping for EML
Capability Transition Timeout subfield value of 11 (indicating 128
TUs).

Signed-off-by: Karthik M <quic_karm@quicinc.com>
Signed-off-by: Yuvarani V <quic_yuvarani@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/wmi.c | 3 +--
 drivers/net/wireless/ath/ath12k/wmi.h | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/wmi.c b/drivers/net/wireless/ath/ath12k/wmi.c
index 07f7477..5a69320 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -198,8 +198,7 @@ static const u32 eml_trans_timeout[EMLCAP_TIMEOUT_MAX] = {
 					EMLCAP_TRANSTIMEOUT_7,
 					EMLCAP_TRANSTIMEOUT_8,
 					EMLCAP_TRANSTIMEOUT_9,
-					EMLCAP_TRANSTIMEOUT_10,
-					EMLCAP_TRANSTIMEOUT_11};
+					EMLCAP_TRANSTIMEOUT_10};
 
 static __le32 ath12k_wmi_tlv_hdr(u32 cmd, u32 len)
 {
diff --git a/drivers/net/wireless/ath/ath12k/wmi.h b/drivers/net/wireless/ath/ath12k/wmi.h
index 9f4e2ea..80eac46 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.h
+++ b/drivers/net/wireless/ath/ath12k/wmi.h
@@ -4393,7 +4393,7 @@ struct wmi_rate_set_arg {
 
 #define EMLSR_DELAY_MAX 5
 #define EMLSR_TRANS_DELAY_MAX 6
-#define EMLCAP_TIMEOUT_MAX 12
+#define EMLCAP_TIMEOUT_MAX 11
 #define IEEE80211_TU_TO_USEC(x)        ((x) << 10)  /* (x)X1024 */
 #define EMLCAP_TRANSTIMEOUT_0  0
 #define EMLCAP_TRANSTIMEOUT_1  (1U << 7)
@@ -4406,7 +4406,6 @@ struct wmi_rate_set_arg {
 #define EMLCAP_TRANSTIMEOUT_8  IEEE80211_TU_TO_USEC((1U << 4))
 #define EMLCAP_TRANSTIMEOUT_9  IEEE80211_TU_TO_USEC((1U << 5))
 #define EMLCAP_TRANSTIMEOUT_10 IEEE80211_TU_TO_USEC((1U << 6))
-#define EMLCAP_TRANSTIMEOUT_11 IEEE80211_TU_TO_USEC((1U << 7))
 
 struct peer_assoc_mlo_params {
 	bool enabled;
-- 
2.17.1

